vitrivr-engine-core/src/main/kotlin/org/vitrivr/engine/core/model/descriptor/struct/MapStructDescriptor.kt

```diff
@@ -0,0 +1,37 @@
+package org.vitrivr.engine.core.model.descriptor.struct
+
+import org.vitrivr.engine.core.model.descriptor.DescriptorId
+import org.vitrivr.engine.core.model.descriptor.Attribute
+import org.vitrivr.engine.core.model.descriptor.AttributeName
+import org.vitrivr.engine.core.model.descriptor.Descriptor
+import org.vitrivr.engine.core.model.metamodel.Schema
+import org.vitrivr.engine.core.model.retrievable.RetrievableId
+import org.vitrivr.engine.core.model.types.Value
+
+/**
+ * An abstract [StructDescriptor] implementation that is backed by a [Map] of [Value]s.
+ *
+ * @author Ralph Gasser
+ * @version 1.0.0
+ */
+open class MapStructDescriptor(
+    override var id: DescriptorId,
+    override var retrievableId: RetrievableId?,
+    protected val layout: List<Attribute>,
+    protected val values: Map<AttributeName, Value<*>?>,
+    override val field: Schema.Field<*, out StructDescriptor>? = null
+) : StructDescriptor {
+    /**
+     * Returns a  [Attribute] [List] for this [Descriptor].
+     *
+     * @return [List] of [Attribute]
+     */
+    final override fun schema(): List<Attribute> = this.layout
```

Comment:
What is the difference between layout and schema?

Reply:
It's a nomenclature issue. We use the word "schema" in two ways:

- To refer to the fields, exporters etc. that make up a schema.
- To refer to the individual attributes that make-up a descriptor.

I'd propose to call the latter "layout", which would require a rename of the method.

Reply:
Makes sense to me, so I'd suggest to do the renaming of the method.

Reply:
Done